vuln-reach/src/javascript/lang/accesses.rs

```diff
@@ -207,6 +215,55 @@ impl<'a> AccessGraph<'a> {
         None
     }
 
+    /// Find the parent scope of a node.
+    fn find_parent_scope(
+        symbol_table: &SymbolTable<'a>,
+        cursor_cache: &mut TreeCursorCache<'a>,
+        node: Node<'a>,
+    ) -> Option<Node<'a>> {
+        let mut parent_cursor = cursor_cache.cursor(node).unwrap();
+        while let Some(node) = parent_cursor.goto_parent() {
+            let parent = parent_cursor.parent()?;
+
+            if parent.child_by_field_name("body") == Some(node)
+                && symbol_table.get_scope(node).is_some()
+            {
+                return Some(node);
+            }
+        }
+        None
+    }
+
+    /// Check where a function is assigned to or executed from.
+    fn find_function_access(
+        symbol_table: &SymbolTable<'a>,
+        cursor_cache: &mut TreeCursorCache<'a>,
+        node: Node<'a>,
+    ) -> Option<Node<'a>> {
+        let mut access_node = None;
+
+        let mut cursor = cursor_cache.cursor(node).unwrap();
+        while let Some(parent) = cursor.goto_parent() {
+            match parent.kind() {
+                // Get identifier if this function is assigned to a variable.
+                "variable_declarator" => access_node = Some(parent),
+                // Find the parent scope for immediately executed anonymous functions.
+                "call_expression" => {
+                    let scope = Self::find_parent_scope(symbol_table, cursor_cache, parent);
+                    access_node = scope?.parent();
+                },
+                // Running into another function without the current one being executed or
+                // assigned means it can never be reached.
```

Comment:
This is not super clear. Is this the correct interpretation? _A function that is immediately surrounded by another function, and is not called, assigned to a variable, or explicitly defined, is unreachable._

```js
function foo() {
  function() { /* this scope is unreachable */ }
}
```

Could we add an example code snippet to clarify, and a test that validates this part?

Reply:
It doesn't need to be immediately surrounded, this works too:

```
function foo() {
  ( ( ( () => { /* unreachable */ } ) ) )
}
```

There also is already a test covering this.